### ISSUE TITLE
Sea ice checkerboarding

### ISSUE THREAD

**Issue by aekiss** [COSIMA/mom6-panan#35]
As mentioned in https://github.com/COSIMA/mom6-panan/issues/34#issuecomment-1487773967, the sea ice concentration displays checkerboarding, indicative of numerical instability. This occurs for DT = 300s in the 1/20th.

Are there SIS2 parameters we can tweak to fix this? It would be good to avoid reducing the timestep if we can.

![Screen Shot 2023-03-29 at Wed 29-3 11 31am](https://user-images.githubusercontent.com/31054815/228396401-a9f7ef4b-48a3-4b21-b6d1-a38091f0d327.png)

![](https://user-images.githubusercontent.com/8506963/228395067-a0d1368d-dac9-45a8-93c7-bb474c4eb1a2.png)

**Comment by aekiss** [comment]
@adele157 what does SIC checkerboard look like with DT = 225?

**Comment by adele157** [comment]
Don't know, it crashed. Is there some way to access the daily output from
`work` of a crashed run?

**Comment by aekiss** [comment]
Not unless there are .nc files you can find in there. Might be enlightening to shorten the run, so it stops just before the crash, and save output from that final state

**Comment by adele157** [comment]
It also occurs in the southern Weddell, but nowhere else. Maybe related to where the grid cells are small?
<img width="1194" alt="Screen Shot 2023-03-30 at 8 29 08 pm" src="https://user-images.githubusercontent.com/8506963/228793188-fd436fa5-19b3-429d-9c6b-90774be10236.png">

**Comment by adele157** [comment]
Ah, and this one that I didn't change going from 1/10 to 1/20:
`DT_ICE_DYNAMICS = 180.0         !   [seconds] default = -1.0`
`                                ! The time step used for the slow ice dynamics, including stepping the`
`                                ! continuity equation and interactions between the ice mass field and`
`                                ! velocities.  `

I had assumed it was the same as the ocean time step! Probably we should halve this going to 1/20th, yes? I'm a bit confused by the large number of different timesteps. I'm thinking they should all be multiples of one another, is that right @angus-g? Otherwise the model adjusts them so they are? There's only a quite small number of possibilities for different timesteps if they all need to be multiples of each other.

**Comment by AndyHoggANU** [comment]
Err. Yes. Good idea to check on the input files! I agree, it should be an integer fraction of the ocean timestep. It probably adjusts to be something like that if it's not, but we clearly need to document how many time steps to change!

**Comment by angus-g** [comment]
> I'm thinking they should all be multiples of one another, is that right @angus-g? Otherwise the model adjusts them so they are?

I don't actually know what happens if they're not, but if it doesn't give a flat-out error it seems like it could manifest as some confusing behaviour. I think this is the difference between the "fast" and "slow" ice timesteps. Maybe it's the "fast" one that's locked to the ocean (dynamics) timestep? I might look for (or create) a diagram relating all the different timesteps we've got going on!

**Comment by adele157** [comment]
There's also:
`DT_RHEOLOGY = 50.0              !   [seconds] default = -1.0`
`                                ! The sub-cycling time step for iterating the rheology and ice momentum`
`                                ! equations.`

and

`NSTEPS_ADV = 1                  ! default = 1`
`                                ! The number of advective iterations for each slow dynamics time step.`

and

**Comment by adele157** [comment]
Which sea ice time step should I try playing with?

1. `DT_ICE_DYNAMICS = 180.0` "for the slow ice dynamics, including stepping the continuity equation and interactions between the ice mass field and velocities". This is equal to 1200 or 3600 in most of the GFDL (coarse-res) examples.
2. or `DT_RHEOLOGY = 50.0` "The sub-cycling time step for iterating the rheology and ice momentum equations." This is equal to 50 or 100 in most GFDL examples.

Also, does anyone know of any high-res MOM6-SIS2 example configs I could compare against?

**Comment by adele157** [comment]
Reducing the timestep gives a big improvement (good one on top, bad one below)! I changed both DT_ICE_DYNAMICS (halved from 180 to 90) and DT_RHEOLOGY (from 50 - which was not a factor of 180 before - to 30). Need to figure out which one solved the problem now. 
<img width="1802" alt="Screen Shot 2023-03-31 at 12 13 35 pm" src="https://user-images.githubusercontent.com/8506963/228998577-41c808a1-1497-459e-b772-b844e55bc5fb.png">

How did we pick `DT_ICE_DYNAMICS = 180.0` @AndyHoggANU ? This seems way lower than in any other examples I can find.

**Comment by AndyHoggANU** [comment]
Not clear to me how we picked `DT_ICE_DYNAMICS` but I would say it is one of those trial and error parameters. If this is the parameter that fixes the checkerboarding then we use that as our metric to set it!

**Comment by aekiss** [comment]
For comparison, CICE5 has 3 timesteps: thermodynamic, dynamic and elastic.

In ACCESS-OM2-01 the sea ice thermodynamic timestep matches the ocean baroclinic timestep, and the CICE5 dynamic timestep is [shorter by a factor of 2](https://github.com/COSIMA/01deg_jra55_iaf/blob/master/ice/cice_in.nml#L25), though we've also experimented with a factor of 3 when hitting CFL limits.

CICE5 uses an elasto-visco-plastic rheology, with an artificial elastic part which creates artificial elastic waves that are supposed to die away during [120 sub-timesteps](https://github.com/COSIMA/01deg_jra55_iaf/blob/master/ice/cice_in.nml#L101) within each dynamic timestep.

I guess SIS2 also uses EVP, and that may be what `DT_RHEOLOGY` is referring to? If so, your `DT_ICE_DYNAMICS`/`DT_RHEOLOGY` = 3 is a lot smaller than the ratio of 120 we use in ACCESS-OM2, and even smaller than the default of 240 in CICE6. This may be what is creating the waves in SIC visible in the first test you did (circled), with a ratio of 3.6:
![Screen Shot 2023-03-31 at Fri 31-3 12 52pm](https://user-images.githubusercontent.com/31054815/229003073-952fdcb5-5d81-431a-9a88-e84c05116b57.png)

**Comment by aekiss** [comment]
What sort of time-averaging was used in the plots so far?

**Comment by adele157** [comment]
They are daily outputs (I assume averages?).

**Comment by aekiss** [comment]
Waves might be more conspicuous in SIS2 diagnostics that look at velocity gradients, eg divergence, strain rate etc.

SIS2 uses the EVP formulation of [Bouillon et al 2009](https://doi.org/10.1016/j.ocemod.2009.01.004) and [Bouillon et al 2013](https://www.sciencedirect.com/science/article/pii/S146350031300098X?via%3Dihub), which follows that in CICE. 

Bouillon et al 2013 used a `DT_ICE_DYNAMICS/DT_RHEOLOGY` ratio of 300.

GFDL OM4p25 and OM4p5 use a rheology time step of 50 s and a dynamics time step of 1200s, ie ratio of 24 ([Adcroft et al 2019](https://doi.org/10.1029/2019MS001726)).

So if we still have elastic waves maybe we should try (say) `DT_RHEOLOGY`=5?

Though oddly, [Bouillon et al 2009](https://doi.org/10.1016/j.ocemod.2009.01.004) say checkerboarding can't occur on a C-grid...

**Comment by adele157** [comment]
For the record, I'm now running the 1/20th with `DT_ICE_DYNAMICS = 225.0`, `DT_RHEOLOGY = 9.0`, based on scaling the GFDL values of these by the resolution factor. This is looking good (from very brief glances at concentration in ncview - this could definitely do with some more people looking in more detail and turning on more sea ice diagnostics!).

**Comment by schmidt-christina** [comment]
I was wondering what other variables to save to investigate this problem further when I run the 1/20th model with the new bathymetry in the Ross Sea.

But before, I just checked Adele's latest run `/home/157/akm157/mom6/panan-005-zstar-ACCESSyr2/` and I still see the above pattern in the first year of her run and also emerging in the second year (not shown).

Do we need to reduce DT even further or save some more diagnostics first to see where it's coming from?

![image](https://user-images.githubusercontent.com/38216156/230265629-c2c4f971-9291-414d-8d96-499c3d4d72c1.png)

![image](https://user-images.githubusercontent.com/38216156/230265816-9a4d6987-1a55-49a0-a3dd-441b2309cc60.png)

**Comment by adele157** [comment]
I only changed the sea ice timesteps in the last run segment I did. I was
waiting for the new bathymetry before running again.

**Comment by schmidt-christina** [comment]
Ok, so this (last day of your run) is ok? There is still a bit of this pattern

![image](https://user-images.githubusercontent.com/38216156/230266648-8307ac96-942e-4bd9-b173-d91c5d49b38b.png)

**Comment by schmidt-christina** [comment]
I can then run the model again with this time step and new Ross Sea bathymetry. Any other sea ice diagnostics I should save?

**Comment by adele157** [comment]
Yeah, looks like most of the checkerboarding is gone, but we still have waves. Is that a problem @aekiss? And how long would we need to run for with the smaller timesteps until these go away? That image above was only a 1 month run with shorter timesteps I think, starting from the bad restart with longer timesteps.

I reckon just start running @schmidt-christina. No doubt something else is going to break and we'll need to restart in any case. We can switch on more diagnostics as we go, but good to just get some more model years under our belt in the meantime, so we can learn a bit more and get the 1/40th going asap.

**Comment by adele157** [comment]
@schmidt-christina can you confirm if the waves disappeared with the change in sea ice timesteps in your last 1/20th run? If so, let's close this.

**Comment by schmidt-christina** [comment]
Yes, there are no waves when running the 1/20th with `DT_ICE_DYNAMICS = 225.0, DT_RHEOLOGY = 9.0`

**Comment by aekiss** [comment]
I think we might need a bigger `DT_ICE_DYNAMICS/DT_RHEOLOGY` ratio than 225/9=25, i.e. a smaller `DT_RHEOLOGY`.

We could use the default `DT_RHEOLOGY = -1.0` to avoid having to adjust `DT_RHEOLOGY` whenever we change `DT_ICE_DYNAMICS`. This will then use the ratio `NSTEPS_DYN` to determine the elastic timestep. `NSTEPS_DYN` defaults to 432, so will give a much smaller `DT_RHEOLOGY` of about 0.52s in our case with `DT_ICE_DYNAMICS = 225.0`.

https://github.com/NOAA-GFDL/SIS2/blob/c565ec390aba0346b26b8c0acedda99386008aad/src/SIS_dyn_cgrid.F90#L204-L218

```fortran
  call get_param(param_file, mdl, "DT_RHEOLOGY", CS%dt_Rheo, &
                 "The sub-cycling time step for iterating the rheology "//&
                 "and ice momentum equations. If DT_RHEOLOGY is negative, "//&
                 "the time step is set via NSTEPS_DYN.", units="seconds", &
                 default=-1.0, scale=US%s_to_T)
  CS%evp_sub_steps = -1
  if (CS%dt_Rheo <= 0.0) &
    call get_param(param_file, mdl, "NSTEPS_DYN", CS%evp_sub_steps, &
                 "The number of iterations of the rheology and ice "//&
                 "momentum equations for each slow ice time step.", default=432)
  call get_param(param_file, mdl, "ICE_TDAMP_ELASTIC", CS%Tdamp, &
                 "The damping timescale associated with the elastic terms "//&
                 "in the sea-ice dynamics equations (if positive) or the "//&
                 "fraction of DT_ICE_DYNAMICS (if negative).", &
                 units="s or nondim", default=-0.2)
```

https://github.com/NOAA-GFDL/SIS2/blob/c565ec390aba0346b26b8c0acedda99386008aad/src/SIS_dyn_cgrid.F90#L776-L781

```fortran
  if (CS%dt_Rheo > 0.0) then
    EVP_steps = max(CEILING(dt_slow/CS%dt_Rheo - 0.0001), 1)
  else
    EVP_steps = CS%evp_sub_steps
  endif
  dt = dt_slow/EVP_steps
```
